scripts: add torch helper

tbg opened the pull request:
This will only work well once we use go1.9, but it will be
helpful then. In the meantime, some of us already use go1.9
locally and can iron out some obvious kinks, such as

https://github.com/uber/go-torch/issues/63
https://github.com/uber/go-torch/issues/61

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17034)
<!-- Reviewable:end -->

bdarnell commented on scripts/torch:
```diff
@@ -0,0 +1,42 @@
+function install() {
+  cat <<EOF
+go get -u github.com/brendangregg/FlameGraph/...
+go get -u github.com/uber/go-torch
```
Should we just put `go-torch` in `Gopkg.toml`? `pprof` is already there.

tschottdorf replied:
Obsolete now that we're using Docker for this helper.

bdarnell commented on scripts/torch:
```diff
@@ -0,0 +1,42 @@
+
+function install() {
+  cat <<EOF
+go get -u github.com/brendangregg/FlameGraph/...
```
Does `go get` not complain about the fact that this repo contains no go code? In any case, it's weird to clone a non-go repo into `GOPATH` (and then to install it onto `$PATH`). I don't have a better idea about where to put it or how to script this, though.

tschottdorf replied:
It does complain but not fail. Anyway, this code is gone.

bdarnell reviewed (approved):
LGTM, but uber also publishes this in a docker image. Would `docker run uber/go-torch` let us avoid the dependency wrangling?

irfansharif commented:
Does vanilla `pprof` work for you? I ran into symbolization issues using this as opposed to `go tool pprof`. From what I can tell it's because `pprof` translates addresses to line information using GNU `addr2line` (can use LLVM `llvm-symbolizer`, see comment [here](https://github.com/google/pprof/issues/130#issuecomment-289539003)) and `go tool pprof` falls back to [`go tool addr2line`](https://golang.org/cmd/addr2line/).

Also given uber/go-torch#64 this might take longer to be useful.

tschottdorf commented:
@irfansharif use go1.9 and `pprof`@master, then everything works (at least for me).

Good point about the Docker image, though they haven't updated it upstream in a while. I suppose we can do that, though.

tbg committed:
scripts: add torch helper

This will only work well once we use go1.9, but it will be
helpful then. In the meantime, some of us already use go1.9
locally and can iron out some obvious kinks, such as

https://github.com/uber/go-torch/issues/63
https://github.com/uber/go-torch/issues/61

tbg reviewed (commented):
@bdarnell I went with your suggestion to use Docker, much more pleasant. I also added a comprehensive usage section. Together with some upstream work, we should be in good shape come 1.9. I smell a lunch and learn.
